### ISSUE TITLE
valuetext doesn't update when formatted value is number-like

### ISSUE THREAD

**Issue by twschiller** [seiyria/bootstrap-slider#799]
When using a formatter that conditionally returns a custom string, or otherwise returns the value as a string, the `aria-valuetext` isn't updated when switching to a value that doesn't use the custom format.

For example, the following formatter:
```
const formatter = v => (p(v) ? custom(v) : v.toString())
```

The cause is that that condition for whether to update `aria-valuetext` uses `isNaN` which returns `true` for strings that are number like, e.g., `"2017"`:

```
if (isNaN(this.options.formatter(this._state.value[0])) ) {
  this.handle1.setAttribute('aria-valuetext', this.options.formatter(this._state.value[0]));
}
```

There are a couple potential fixes:
1. Unset the value of `aria-valuetext` so there's no outdated information when the formatter returns a number-like value
2. Change the condition, e.g., to `typeof <formatted> === "string"` so that the text is always updated for formatters that return strings

Option 1 probably the better fix, unless you require that formatter always require a `string` return value

**Comment by jespirit** [comment]
I was reading the MDN docs of the `aria-valuetext` attribute and the code seems to have it correct. The `aria-valuetext` _should_ only be set when the values of the slider should be rendered as human readable text.

For example, if you represent the days of a week as numbers from 1-7, `aria-valuenow` should be a numeric value from 1-7, but the `aria-valuetext` should be set to a human-readable string (eg. "Sunday") which the user agent should render instead.

So it only makes sense to set the `aria-valuetext` attribute when the formatted value is not a number.

[Using the aria-valuetext attribute
](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-valuetext_attribute)

**Comment by twschiller** [comment]
Thanks for investigating @jespirit. IIRC, the issue we had is if you switch from a string, e.g., "2018 YTD" to "2017" the valuetext won't update. Based on the Mozilla documentation, seems that the correct behavior would be to explicitly unset the valuetext
